plutus-scb-client/src/View.purs

```diff
@@ -0,0 +1,235 @@
+import Plutus.SCB.Webserver.Types (FullReport(..))
+import Prelude (class Eq, class Show, otherwise, show, ($), (+), (<$>), (<<<), (<>), (==))
+import Types (HAction(..), State(State), _hooks, _partiallyDecodedResponse)
+import Wallet.Emulator.Wallet (Wallet)
+import Wallet.Rollup.Types (AnnotatedTx)
+
+render ::
+  forall m slots.
+  MonadAff m =>
+  State -> ComponentHTML HAction slots m
+render (State { chainState, fullReport }) =
+  div
+    [ class_ $ ClassName "main-frame" ]
+    [ container_
+        [ div_
+            $ case fullReport of
+                Success report -> [ fullReportPane chainState report ]
+                Failure error -> [ ajaxErrorPane error ]
+                Loading -> [ icon Spinner ]
+                NotAsked -> [ icon Spinner ]
+        ]
+    ]
+
+fullReportPane :: forall p. Chain.State -> FullReport -> HTML p HAction
+fullReportPane chainState fullReport@(FullReport { events, latestContractStatus, transactionMap, utxoIndex, annotatedBlockchain, walletMap }) =
+  row_
+    [ col10_ [ contractStatusPane latestContractStatus ]
+    , col12_ [ ChainAction <<< Just <$> annotatedBlockchainPane chainState walletMap annotatedBlockchain ]
+    , col12_ [ eventsPane events ]
+    , col8_ [ transactionPane transactionMap ]
+    , col8_ [ utxoIndexPane utxoIndex ]
+    ]
+
+contractStatusPane :: forall p i. JsonMap JsonUUID ActiveContractState -> HTML p i
+contractStatusPane latestContractStatus =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Active Contracts" ]
+        ]
+    , cardBody_
+        [ div_
+            ( ( \(Tuple k v) ->
+                  row_
+                    [ col3_ [ h3_ [ text $ view (_JsonUUID <<< to UUID.toString) k ] ]
+                    , col9_ [ text $ view (_partiallyDecodedResponse <<< _hooks <<< _RawJson) v ]
+                    ]
+              )
+                <$> (Map.toUnfoldable $ unwrap latestContractStatus :: Array (Tuple JsonUUID ActiveContractState))
+            )
+        ]
+    ]
+
+annotatedBlockchainPane :: forall p. Chain.State -> JsonMap PubKeyHash Wallet -> Array (Array AnnotatedTx) -> HTML p ChainFocus
+annotatedBlockchainPane chainState walletMap chain =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Blockchain" ]
+        ]
+    , cardBody_
+        [ chainView chainState (unwrap walletMap) $ AnnotatedBlockchain chain
+        ]
+    ]
+
+transactionPane ::
+  forall p i.
+  JsonMap TxId Tx -> HTML p i
+transactionPane txMap =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Txs" ]
+        ]
+    , cardBody_
+        ( toArrayOf
+            ( _JsonMap
+                <<< traversed
+                <<< to (\x -> div_ [ code_ [ text $ show x ] ])
+            )
+            txMap
+        )
+    ]
+
+utxoIndexPane :: forall p i. UtxoIndex -> HTML p i
+utxoIndexPane utxoIndex =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "UtxoIndex" ]
+        ]
+    , cardBody_ [ div_ [ code_ [ text $ show utxoIndex ] ] ]
+    ]
```

Comment:
`(UtxoIndex { getIndex: (JsonMap (fromFoldable [])) })`

Looks a bit rubbish, could at least be `empty`. I presume non-empty ones will look a bit bad as raw JSON also. Lol, I don't mean to sound nasty :P

Reply:
On this one I think I'll push back. There's lots to be done to improve the look. This was really meant to be a "get the right stuff on the page (and get an actual page)" commit. Build issues have left this PR alive much longer than I expected.

Reply:
ok